sdk/src/Services/DynamoDBv2/Custom/DataModel/_async/Context.Async.cs

```diff
@@ -412,19 +436,41 @@ public AsyncSearch<T> QueryAsync<T>(object hashKeyValue, QueryOperator op, IEnum
             return FromSearchAsync<T>(query);
         }
 
-        /// <summary>
-        /// Configures an async Query operation against DynamoDB, finding items
-        /// that match the specified conditions.
-        /// </summary>
-        /// <typeparam name="T">Type of object.</typeparam>
-        /// <param name="queryConfig">Query request object.</param>
-        /// <param name="operationConfig">Config object which can be used to override the table used.</param>
-        /// <returns>AsyncSearch which can be used to retrieve DynamoDB data.</returns>
-        public AsyncSearch<T> FromQueryAsync<T>(QueryOperationConfig queryConfig, DynamoDBOperationConfig operationConfig = null)
+        /// <inheritdoc/>
+        public AsyncSearch<T> QueryAsync<T>(object hashKeyValue, QueryOperator op, IEnumerable<object> values, QueryConfig queryConfig)
+        {
+            if (values == null)
+                throw new ArgumentNullException("values");
+
+            var query = ConvertQueryByValue<T>(hashKeyValue, op, values, queryConfig?.ToDynamoDBOperationConfig());
+            return FromSearchAsync<T>(query);
+        }
+
+        /// <inheritdoc/>
+        public AsyncSearch<T> FromQueryAsync<T>(QueryOperationConfig queryOperationConfig)
+        {
+            if (queryOperationConfig == null) throw new ArgumentNullException("queryOperationConfig");
+
+            var search = ConvertFromQuery<T>(queryOperationConfig, null);
+            return FromSearchAsync<T>(search);
+        }
+
+        /// <inheritdoc/>
+        [Obsolete("Use the FromQueryAsync overload that takes QueryConfig instead, since DynamoDBOperationConfig contains properties that are not applicable to FromQueryAsync.")]
+        public AsyncSearch<T> FromQueryAsync<T>(QueryOperationConfig queryOperationConfig, DynamoDBOperationConfig operationConfig = null)
```

Comment:
Wouldn't renaming the argument from `queryConfig` -> `queryOperationConfig` be a breaking change?

Reply:
Reset in https://github.com/aws/aws-sdk-net/pull/3383/commits/987fe702d9428657cff4228bf91457a1ae9dcd74, and names are now less ambiguous after the adjustments due to this thread: https://github.com/aws/aws-sdk-net/pull/3383#discussion_r1673120405